File: console/src/utils/token.ts
import Cookies from "js-cookie";

export const setToken = (token: string) => {
  Cookies.set("access_token", token);
};

export const getToken = () => {
  const token = Cookies.get("access_token");
  return token;
};

export const setRefreshToken = (token: string) => {
  Cookies.set("refresh_token", token);
};

export const getRefreshToken = () => {
  const token = Cookies.get("refresh_token");
  return token;
};

File: brume/internal/router/http/cookies/token.go
package cookies

import (
	"net/http"
	"time"

	"brume.dev/internal/config"
)

func GenerateUserCookie(accessToken string, refreshToken string, cfg *config.BrumeConfig, w http.ResponseWriter) *http.Cookie {
	http.SetCookie(w, &http.Cookie{
		Name:     "access_token",
		Value:    accessToken,
		Path:     "/",
		Expires:  time.Now().Add(4 * time.Hour),
		HttpOnly: true,
		Secure:   false,
		SameSite: http.SameSiteLaxMode,
	})
	http.SetCookie(w, &http.Cookie{
		Name:     "refresh_token",
		Value:    refreshToken,
		Path:     "/",
		HttpOnly: true,
		Secure:   false,
		SameSite: http.SameSiteLaxMode,
	})
	return nil
}
